app/code/Magento/Backend/Model/Url.php

```diff
@@ -166,7 +167,10 @@ protected function _isSecure()
         if ($this->hasData('secure_is_forced')) {
             return $this->getData('secure');
         }
-        return $this->_scopeConfig->isSetFlag('web/secure/use_in_adminhtml');
+        if ($this->_scopeConfig->isSetFlag('web/secure/use_in_adminhtml')) {
+            return $this->getData('secure');
+        }
+        return parent::_isSecure();
```

Comment:
This is a _really_ risky fallback. Please provide exact scenario you trying to fix with it, as now it is pretty easy to get a non-secure admin open for MITM attacks and such.

Reply:
I iam not going to provide any more scenarios.. now there is a vulnerability in magento, and this fixes it.. Please someone other make look at it or think again.. if you are not wise ehough to understand why it should be merged to magento, i am very disappointed on how magento picks human resources :(